src/TraceEvent/TraceLoggingEventId.cs

```diff
@@ -38,8 +40,8 @@ public void Dispose()
             if (m_traceLoggingEventMap == null)
                 return;
 
-            foreach (var key in m_traceLoggingEventMap.Keys)
-                Marshal.FreeHGlobal((IntPtr)key.Provider);
+            foreach (var kvp in m_traceLoggingEventMap)
+                Marshal.FreeHGlobal((IntPtr)kvp.Key.Provider);
```

Comment:
:memo: For those unaware, this change eliminates an unnecessary allocation during the `Dispose` call. One of the things I'm enjoying about C# 7 (which we aren't currently using on this project) is the ability to gain the efficiency of this form without losing the clarity of the original code by writing this:

```csharp
foreach (var (key, _) in m_traceLoggingEventMap)
  Marshal.FreeHGlobal((IntPtr)key.Provider);
```

This form applies a deconstruction to the `KeyValuePair<TKey, TValue>` in the declaration of `key` and an unused placeholder `_`.